rust/worker/src/blockstore/arrow/sparse_index.rs

```diff
@@ -117,6 +117,131 @@ impl SparseIndex {
         panic!("No blocks in the sparse index");
     }
 
+    pub(super) fn get_block_ids_prefix(&self, prefix: &str) -> Vec<Uuid> {
+        let lock_guard = self.forward.lock();
+        let mut curr_iter = lock_guard.iter();
+        let mut next_iter = lock_guard.iter().skip(1);
+        let mut block_ids = vec![];
+        while let Some((curr_key, curr_uuid)) = curr_iter.next() {
+            let non_start_curr_key: Option<&CompositeKey>;
+            match curr_key {
+                SparseIndexDelimiter::Start => non_start_curr_key = None,
+                SparseIndexDelimiter::Key(k) => non_start_curr_key = Some(k),
+            }
+            if let Some((next_key, _)) = next_iter.next() {
+                // This can't be a start key but we still need to extract it.
+                let non_start_next_key: Option<&CompositeKey>;
+                match next_key {
+                    SparseIndexDelimiter::Start => {
+                        panic!("Invariance violation. Sparse index is not valid.");
+                    }
+                    SparseIndexDelimiter::Key(k) => non_start_next_key = Some(k),
+                }
+                // If delimeter starts with the same prefix then there will be keys inside the
+                // block with this prefix.
+                if non_start_curr_key.is_some()
+                    && prefix == non_start_curr_key.clone().unwrap().prefix.as_str()
+                {
+                    block_ids.push(*curr_uuid);
+                }
+                // If prefix is between the current delim and next delim then there could
+                // be keys in this block that have this prefix.
+                if (non_start_curr_key.is_none()
+                    || prefix > non_start_curr_key.unwrap().prefix.as_str())
+                    && (prefix <= non_start_next_key.unwrap().prefix.as_str())
+                {
+                    block_ids.push(*curr_uuid);
+                }
+            } else {
+                // Last block.
+                if prefix >= non_start_curr_key.unwrap().prefix.as_str() {
+                    block_ids.push(*curr_uuid);
+                }
+            }
+        }
+        block_ids
+    }
+
+    pub(super) fn get_block_ids_gt(&self, search_key: CompositeKey) -> Vec<Uuid> {
```

Comment:
Given my above comment and understanding of `gt`, `gte`, `lt`, and `lte`, I think we can get rid of this since we only ever need exact prefix matching.

Reply:
You're right, we do still need this. But like the blockfile lookup it should do exact prefix comparison and inequality comparison only on keys